ndb_api/common_helpers.go

```diff
@@ -49,3 +51,91 @@ func GetDatabasePortByType(dbType string) int32 {
 		return -1
 	}
 }
+
+// Sorts action arguments in place by name
+func sortActionArgsByName(actionArgs []ActionArgument) {
+	sort.Slice(actionArgs, func(i, j int) bool {
+		return actionArgs[i].Name < actionArgs[j].Name
+	})
+}
+
+// Gets MSSQL overridable action arguments.
+func getMsSQLOverridableActionArgs(dbParameterProfileIdInstance string, vmDbServerAdminPassword string) []ActionArgument {
+	return []ActionArgument{
+		{Name: "sql_user_name", Value: "sa"},
+		{Name: "authentication_mode", Value: "windows"},
+		{Name: "server_collation", Value: "SQL_Latin1_General_CP1_CI_AS"},
+		{Name: "database_collation", Value: "SQL_Latin1_General_CP1_CI_AS"},
+		{Name: "dbParameterProfileIdInstance", Value: dbParameterProfileIdInstance},
+		{Name: "vm_dbserver_admin_password", Value: vmDbServerAdminPassword},
+	}
+}
+
+// Gets MSSQL non-overridable action arguments.
+func getMsSQLNonOverridableActionArgs(dbServerName string) []ActionArgument {
+	return []ActionArgument{
+		{Name: "working_dir", Value: "C:\\temp"},
+		{Name: "delete_vm_on_failure", Value: "false"},
+		{Name: "is_gmsa_sql_service_account", Value: "false"},
+		{Name: "provision_from_backup", Value: "false"},
+		{Name: "distribute_database_data", Value: "true"},
+		{Name: "retain_database_in_restoring_mode", Value: "false"},
+		{Name: "dbserver_name", Value: dbServerName},
+	}
+}
```

Comment:
Whats the difference between overridable, non-overridable and user provided args?

Reply:
NonOverridable is NOT specifiable by the user. 
Overridable  is specifiable.

I changed how I do this per your comment, see the PR description for an explanation.